src/components/SignerPanel/ActionPathsContent.js

```diff
@@ -24,13 +30,56 @@ class ActionPathsContent extends React.Component {
       pretransaction
     )
   }
-  renderDescription(showPaths, { description, name, to }) {
+  renderDescription(
+    showPaths,
+    { description, name, to, annotatedDescription }
+  ) {
     return (
       <React.Fragment>
-        <p>This transaction will {showPaths && 'eventually'} perform:</p>
-        <p style={{ margin: '10px 0' }}>
-          {description ? `"${description}"` : 'an action'}
-        </p>
+        <p>This transaction will {showPaths && 'eventually'} perform</p>
+        <div style={{ margin: '10px 0 10px 15px' }}>
+          {annotatedDescription
+            ? annotatedDescription.map(({ type, value }, index) => {
+                if (type === 'address') {
+                  return (
+                    <IdentityBadge
+                      key={index}
+                      entity={value}
+                      fontSize="small"
+                      style={{ verticalAlign: 'middle', marginRight: '4px' }}
+                    />
+                  )
+                } else if (type === 'app') {
+                  return (
+                    <SafeLink
+                      href={`/#/${
+                        this.props.locator.dao
+                      }/permissions/?params=app.${value.proxyAddress}`}
+                      target="_blank"
```

Comment:
Is this meant to open a new tab?

cc @bpierre In the future this should use the deep linking / navigation API we provide.

Reply:
I think it makes sense this opens in a new tab actually. Taking the user out of their flow to go to the permissions app would be weird.